### PR TITLE
Prometheus Receiver metric type fixes to match Prometheus functionality

### PR DESCRIPTION
**Pull request by gracewehner**
**Description:** 
Same as https://github.com/open-telemetry/opentelemetry-collector/pull/3853 but moved here with Prometheus receiver.

I also addressed the feedback to also make changes to otlp_metricfamily.go for when we switch over to those files. The current files are not caught up with what's in metricfamily.go and https://github.com/open-telemetry/opentelemetry-collector/pull/3741, so some merge conflicts may need to be fixed when https://github.com/open-telemetry/opentelemetry-collector/pull/3741 is merged (like uncommenting the logger line).

1. The prometheus receiver sets metrics without the `# TYPE` hint as UNSPECIFIED which converts to empty OTLP metrics: such as `{"metrics":[{},{},{}]}` in the OTLP json output and considers these metric invalid. Prometheus, however, will treat these metrics as a Gauge type and these metrics show up correctly in the Prometheus expression browser. With these changes, if the metadata type is not found, the type is set to `textparse.MetricTypeUnknown` so that in the function `convToOCAMetricType(metadata.Type)`, the type is converted to `metricspb.MetricDescriptor_GAUGE_DOUBLE`.
2. https://github.com/open-telemetry/opentelemetry-collector/issues/3557 was created since some counter metrics sent when using the python prometheus sdk were being considered invalid metrics. This is because in the Prometheus metadata lookup, their name excluded the _total, so their metadata couldn't be found. https://github.com/open-telemetry/opentelemetry-collector/pull/3603 fixed the issue by checking the name without the _total suffix to lookup the metadata. However, the family name without the _total suffix is used as the metric name, so the metric `python_gc_objects_collected_total` is converted to the name `python_gc_objects_collected` whereas Prometheus still includes the _total suffix. This is because previously the metric family name only differed from metric names when the metric was a summary and histogram that have the metrics with _sum and _count, which are added back in later. There isn't this special logic for counters, so the family name needs to be set to the actual metric name when it is a counter and has the _total suffix.

**Link to tracking Issue:** #4743 and https://github.com/open-telemetry/opentelemetry-collector/issues/3557

**Testing:**
Tested out on a Kubernetes cluster with Prometheus scrape targets that have metrics without the `# TYPE` hint for (1) and metrics using the python sdk sending metrics such as below for testing (2):
```
python_gc_objects_collected_total
python_gc_objects_uncollectable_total 
python_gc_collections_total
process_cpu_seconds_total
```
Changed tests for metrics without the type hint to expect the gauge type.
Added test case for situation where metadata key excludes _total suffix but metric name should still have _total suffix

### CODE REVIEW

File: receiver/prometheusreceiver/internal/otlp_metricfamily.go [hunk 1]
```diff
@@ -67,10 +67,22 @@ func newMetricFamilyPdata(metricName string, mc MetadataCache, intervalStartTime
 			metadata.Metric = familyName
 			metadata.Type = textparse.MetricTypeUnknown
 		}
+	} else if !ok {
+		// Prometheus sends metrics without a type hint as gauges.
+		// MetricTypeUnknown is converted a gauge in convToOCAMetricType()
+		//logger.Debug(fmt.Sprintf("Metadata unknown, using Gauge type for: %s %+v", metricName, metadata))
```
Comment (dashpole) [inline receiver/prometheusreceiver/internal/otlp_metricfamily.go:73]: nit: We should either uncomment the debug statement, to match the OC metricfamily change, or remove both debug statements (this and the opencensus one above)

**Review by dashpole — APPROVED** [review]
LGTM with one nit.

**Comment by gracewehner** [comment]
Hi @Aneurysm9 could you please take a look at this PR? We would like to get this in with the upcoming release. Thank you!

**Review by Aneurysm9 — APPROVED** [review]

**Comment by gracewehner** [comment]
Hi @bogdandrutu could you please take a look for this to be merged? Thank you!

**Comment by gracewehner** [comment]
Hi @alolita could this PR be marked ready for merge? We would like to get this in for the next release. Thank you!

**Pull request merged**